futures-executor/src/task_runner.rs

```diff
@@ -298,6 +298,7 @@ impl Future for SpawnedFuture {
     type Error = bool;
 
     fn poll(&mut self, ctx: &mut task::Context) -> Poll<bool, bool> {
-        self.inner.with_task_data(|f| f.poll(ctx))
+        //self.inner.with_task_data(|f| f.poll(ctx))
+        self.inner.get_mut().poll(ctx)
```

Comment:
I'm not sure I understand why this was implemented this way to begin with. It seems like it would be a noop, but I have no idea where the current task gets set.